pictview: added support for viewing .heif, .webp and .xpm files

lejcik opened the pull request:
Added experimental support for viewing `.heif` and `.webp` image files into PictView plugin.

For more details see the discussion here: https://github.com/OpenSalamander/salamander/discussions/36

**NOTE:** This feature is not ready to be merged, and that this pull request will stay as a draft.

lejcik committed:
pictview: added support for viewing .heif and .webp files (experimental)

lejcik committed:
cast correction

lejcik committed:
a few more improvements for pict-view overrider

lejcik committed:
minor cleanup

lejcik committed:
use vcpkg manifest file

lejcik committed:
clang-formated PVW32DLL.h

melloware commented:
@lejcik you are my hero ❤️

melloware commented:
cc @WaldemarH would love your eyes on this and any opinion?  The PictView plugin is one of the most useful and popular plugins!

lejcik commented:
it will take a while to set up automated builds, but those who want to try out the updated PictView plugin, you can download it from here: https://uschovna.cz/zasilka/RG2D5WT7IRNEC89S-XS5

melloware commented:
I just tried the x64 version and I keep getting this...

![image](https://github.com/user-attachments/assets/464561e6-7a6e-467f-ab16-7e2a194aed2c)

lejcik commented:
strange, on my PC it works well,

ok, firstly go to menu plugins/plugin manager and try to test this plugin, if it fails, try to load other plugins if possible (just to test if you have the VC redistributables installed on your PC),

if it still fails to load, try to use [procmon tool](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon), and watch what dlls is salamander trying to load, and which dll it can't find,

I may try out to run this build on a clean system on virtual machine, but I don't have time for that now...

ags1234 commented:
Ok, downloaded Open Salamander 5.0 x64.
No dll errors here, but webp files are displayed as hex, not as picture.

melloware commented:
I confirm I can load other plugins fine but the PictView plugin hangs for a few seconds and then displays the errors above.  

![image](https://github.com/user-attachments/assets/bf8e91fe-f393-4d9d-8c57-0251e4931999)

melloware commented:
I assume my issue is I don't have the VC++ Redistributable of the version you are building with?

ags1234 commented:
@melloware 
If missing any VC++ redistributable: https://github.com/abbodi1406/vcredist

lejcik commented:
> But webp files are displayed as hex, not as picture.

this may happen if salamander configuration was imported from previous version, but if you load the plugin (e.g. via plugin manager, click on test button), it will update salamander configuration, and register new file extensions,

also test if thumbnails are displayed well,

melloware commented:
OK i did do import my configuration when it asked me how do I back that out?

ags1234 commented:
@lejcik 
Salamander 5 offered to import config from Salamander 4 but i refused, and checked to delete it.
I also clicked on Test button, and also restarted OS 5, but pictures still show as hex.

melloware commented:
OK confirmed the VC++ issue that @ags1234 posted fixed me up and .WEBP show up fine for me!

lejcik commented:
> ... but webp pictures still show as hex.

check if file extension is registered in salamander, go to menu options/configuration/viewers and there's a list of file extensions, if there's no `*.webp`, you can add it to the list and assign it to PictView plugin, but plugin should register it once it is loaded,

ags1234 commented:
> > ... but webp pictures still show as hex.
> 
> check if file extension is registered in salamander, go to menu options/configuration/viewers and there's a list of file extensions, if there's no `*.webp`, you can add it to the list and assign it to PictView plugin, but plugin should register it once it is loaded,

Registered *.webp and associated it with PictView (Plugin). Even closed OS and reopened and still hex on any webp files.
The following work fine wirh F3: jpg, png, gif.
There's still something in the code of PictView that makes it not recognize webp as a picture, but as a bniary file.

melloware commented:
Just confirming I went here: https://developers.google.com/speed/webp/gallery1  downloaded an example webp file and its opens with F3 for me.  Could it be that particular WEBP you are trying to open?

tormento commented:
I found a build of AS 5. Please share your plugin ;)

lejcik commented:
ok, here is the pictview plugin for salamander 5:

- [pictview-x86.zip](https://github.com/user-attachments/files/22215558/pictview-x86.zip)
- [pictview-x64.zip](https://github.com/user-attachments/files/22215559/pictview-x64.zip)

the install instructions stay the same as in my comment above... does the plugin work well?

lejcik committed:
Merge branch 'main' into plugin-pictview-support-for-heif-and-webp

lejcik committed:
removed redundant call

lejcik committed:
heif: show progress and allow cancellation during picture loading

some picture decoders that are supported by libheif ignore the
heif_decoding_options, and that we can't get loading progress and
we can't cancel the loading procedure!

lejcik committed:
added support for viewing xpm image files

lejcik commented:
a small update - added support for viewing .xpm files

lejcik committed:
Merge branch 'main' into plugin-pictview-support-for-heif-and-webp

lejcik committed:
Merge branch 'main' into plugin-pictview-support-for-heif-and-webp

lejcik committed:
Merge branch 'main' into plugin-pictview-support-for-heif-and-webp
